File: client/result_tree_handler.go
package client

import (
	proto "github.com/rgraphql/rgraphql"
)

// ResultTreeHandler handles a result value sequence.
type ResultTreeHandler interface {
	// HandleResultValue handles the next value in the sequence, optionally
	// returning a handler for the next value(s) in the sequence.
	HandleResultValue(val *proto.RGQLValue) ResultTreeHandler
}

File: client/result-tree-handler.ts
// HandleResultValue handles the next value in the sequence, optionally
// returning a handler for the next value(s) in the sequence.

import { RGQLValue } from '../rgraphql.pb.js'

// If val == undefined, delete the value.
export type ResultTreeHandler = ((val: RGQLValue | undefined) => ResultTreeHandler) | null
